src/ckeditor.jsx

```diff
@@ -230,16 +231,23 @@ export default class CKEditor extends React.Component {
 	async _destroyEditor() {
 		this.editorDestructionInProgress = new Promise( resolve => {
 			// It may happen during the tests that the watchdog instance is not assigned before destroying itself. See: #197.
+			//
+			// Additionally, we need to find a way to detect if the whole context has been destroyed. As `componentWillUnmount()`
+			// could be fired by <CKEditorContext /> and <CKEditor /> at the same time, this `setTimeout()` makes sure
+			// that <CKEditorContext /> component will be destroyed first, so during the code execution
+			// the `ContextWatchdog#state` would have a correct value. See `EditorWatchdogAdapter#destroy()` for more information.
 			/* istanbul ignore next */
-			if ( this.watchdog ) {
-				this.watchdog.destroy().then( () => {
-					this.watchdog = null;
-
+			setTimeout( () => {
+				if ( this.watchdog ) {
+					this.watchdog.destroy().then( () => {
+						this.watchdog = null;
+
+						resolve();
+					} );
+				} else {
 					resolve();
-				} );
-			} else {
-				resolve();
-			}
+				}
+			} );
```

Comment:
No idea how to get rid of this `setTimeout()`, as the `<CKEditor />` does not know if `<CKEditorContex />` is going to destroy the editor.

Let's keep it as it is.